sources/lib/structures/ty.mli

```diff
@@ -26,77 +26,204 @@
 (*                                                                            *)
 (******************************************************************************)
 
+(** Types
+
+    This module defines the representation of types. *)
+
+(** {2 Definition} *)
+
 type t =
   | Tint
+  (** Integer numbers *)
   | Treal
+  (** Real numbers *)
   | Tbool
+  (** Booleans *)
   | Tunit
+  (** The unit type *)
   | Tvar of tvar
+  (** Type variables *)
   | Tbitv of int
+  (** Bitvectors of a given length *)
   | Text of t list * Hstring.t
+  (** Abstract types applied to arguments. [Text (args, s)] is
+      the application of the abstract type constructor [s] to
+      arguments [args]. *)
   | Tfarray of t * t
+  (** Functional arrays. [TFarray (src,dst)] maps values of type [src]
+      to values of type [dst]. *)
   | Tnext of t
+  (** TODO: ? *)
   | Tsum of Hstring.t * Hstring.t list
+  (** Enumeration, with its name, and the list of its constructors. *)
   | Trecord of trecord
+  (** Record type. *)
+
+and tvar = {
+  v : int;
+  (** Unique identifier *)
+  mutable value : t option;
+  (** Pointer to the current value of the type variable. *)
+}
+(** Type variables.
+    The [value] field is mutated during unification,
+    hence distinct types should have disjoints sets of
+    type variables (see function {fresh}). *)
 
-and tvar = { v : int ; mutable value : t option }
 and trecord = {
   mutable args : t list;
+  (** parameters of the record
+      TODO: or are they the arguments applied
+            to the record constructor type ? *)
   name : Hstring.t;
-  mutable lbs :  (Hstring.t * t) list
+  (** Name of the record type *)
+  mutable lbs :  (Hstring.t * t) list;
+  (** List of fields of the record. Each field has a name,
+      and an associated type. *)
 }
+(** Record types. *)
 
+module Svty : Set.S with type elt = int
+(** Sets of type variables *)
 
-module M : Map.S with type key = int
+module Set : Set.S with type elt = t
+(** Sets of types *)
 
-type subst = t M.t
 
-val esubst : subst
+(** {2 Type inspection} *)
+
+val hash : t -> int
+(** Hash function *)
+
+val equal : t -> t -> bool
+(** Equality function *)
+
+val compare : t -> t -> int
+(** Comparison function *)
+
+val print : Format.formatter -> t -> unit
+(** Printing function for types (does not print
+    the type of each fields for records). *)
+
+val print_list : Format.formatter -> t list -> unit
+(** Print function for lists of types (does not print
+    the type of each fields for records). *)
+
+val print_full : Format.formatter -> t -> unit
+(** Print function including the record fields. *)
```

Comment:
and the constructors of enumerations I think

Reply:
After looking at the code, the contructors of an enumerations are never printed currently.

Some justification for that would be that, these constructors are always the same, contrary to record fields, whose actual type may change due to type variable substitution.